Mathlib/NumberTheory/Cyclotomic/PrimitiveRoots.lean

```diff
@@ -189,6 +189,85 @@ theorem finrank (hirr : Irreducible (cyclotomic n K)) : finrank K L = (n : ℕ).
     (zeta_spec n K L).minpoly_eq_cyclotomic_of_irreducible hirr, natDegree_cyclotomic]
 #align is_cyclotomic_extension.finrank IsCyclotomicExtension.finrank
 
+variable {L} in
+/-- If `L` contains both a primitive `p`-th root of unity and `q`-th root of unity, and
+`Irreducible (cyclotomic (lcm p q) K)` (in particular for `K = ℚ`), then the `finrank K L` is at
+least `(lcm p q).totient`. -/
+theorem _root_.IsPrimitiveRoot.lcm_totient_le_finrank [FiniteDimensional K L] {p q : ℕ} {x y : L}
```

Comment:
I think it makes sense to end the namespace `IsCyclotomicExtension` before your additions and change to `namespace IsPrimitiveRoot`, so as to avoid having to prefix everything with `_root_.IsPrimitiveRoot.`.

Reply:
I've moved the results to the following namespace.